### ISSUE TITLE
`standards_conforming_strings=off` confuses duckdb parser

### ISSUE THREAD

**Issue by hlinnaka** [duckdb/pg_duckdb#119]
```
postgres=# set standard_conforming_strings =off;
SET
postgres=# select * from foo where t = 'foo\'bar';
WARNING:  nonstandard use of \' in a string literal
LINE 1: select * from foo where t = 'foo\'bar';
                                    ^
HINT:  Use '' to write quotes in strings, or use the escape string syntax (E'...').
WARNING:  (DuckDB) Parser Error: syntax error at or near "bar"
LINE 1: select * from foo where t = 'foo\'bar';
                                          ^
ERROR:  unrecognized node type: 1
```

**Comment by Reminiscent** [comment]
@wuputah @hlinnaka What do you think the best way to fix this issue?
1. support `standards_conforming_strings` in `duckdb` for compatible.
2. handle this logic in the pg_duckdb. Do some pre-process in the pg_duckdb.
3. forbid to use the duckdb when we find we use this GUC.

A general question is what is the general approach for some GUCs that may change behavior or be incompatible with duckdb, or is it handled on a case-by-case?

**Comment by JelteF** [comment]
This is now fixed by #130, I now added a test for this in #135
